saml.go

```diff
@@ -219,6 +219,7 @@ type WarningInfo struct {
 }
 
 type AssertionInfo struct {
+	UserName    string
```

Comment:
IMO this field should be called `NameID` instead of trying to abstract it. The spec describes the value as potentially "opaque" so it might not be a user name at all.